src/main/java/org/jenkinsci/deprecatedusage/JavadocUtil.java

```diff
@@ -0,0 +1,135 @@
+package org.jenkinsci.deprecatedusage;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class JavadocUtil {
+
+    private static final String JAVADOC_URL = "http://javadoc.jenkins.io/";
+
+    public static String signatureToJenkinsdocLink(String fullSignature) {
+        return signatureToJenkinsdocLink(fullSignature, fullSignature);
+    }
+
+    public static String signatureToJenkinsdocLink(String fullSignature, String label) {
+        String url = signatureToJenkinsdocUrl(fullSignature);
+
+        label = label.replace("<", "&lt;").replace(">", "&gt;");
+
+        if (!fullSignature.contains("jenkins") && !fullSignature.contains("hudson")) {
+            return label;
+        }
+
+        return "<a href='" + url+ "'>" + label + "</a>";
+    }
+
+    public static String signatureToJenkinsdocUrl(String fullSignature) {
+
+        boolean isClass = !fullSignature.contains("#");
+        boolean isField = !isClass && !fullSignature.contains("(");
+        boolean isMethod = !isClass && fullSignature.contains("(");
+
+        if (isClass) {
+            // transform package and class names, then return
+            return JAVADOC_URL + fullSignature.replace("$", ".") + ".html";
+        }
+
+        String packageName = "";
+        String classMethodArgumentsAndReturn = fullSignature;
+        String packageAndClass = fullSignature.substring(0, fullSignature.indexOf("#"));
+
+        int endOfPackage =  packageAndClass.lastIndexOf("/");
+        if (endOfPackage > 0) {
+            packageName = fullSignature.substring(0, endOfPackage);
+            classMethodArgumentsAndReturn = fullSignature.substring(endOfPackage + 1);
+        }
+
+        if (isField) {
+            String className = fullSignature.substring(0, fullSignature.indexOf("#"));
```

Comment:
Yeah, this insanity took a few iterations to get right, and only discovered very late how easy the package and class are to transform.